### PR TITLE
CUDA 12.0 conda support

### PR DESCRIPTION
**Pull request by bdice**
This PR enables CUDA 12.0 workflows for building RAPIDS conda packages. This PR is planned to merge on July 18, 2023.

### CODE REVIEW

**Comment by vyasr** [comment]
This PR is so deceptively tiny for all the work that's gone into CUDA 12 support :joy:

File: .github/workflows/conda-cpp-tests.yaml [hunk 1]
```diff
@@ -65,7 +65,7 @@ jobs:
               - { CUDA_VER: '11.2.2', LINUX_VER: 'centos7', ARCH: 'amd64', PY_VER: '3.9', GPU: 'v100', DRIVER: 'earliest' }
               - { CUDA_VER: '11.4.3', LINUX_VER: 'ubuntu20.04', ARCH: 'amd64', PY_VER: '3.9', GPU: 'v100', DRIVER: 'latest' }
               - { CUDA_VER: '11.8.0', LINUX_VER: 'ubuntu22.04', ARCH: 'amd64', PY_VER: '3.10', GPU: 'v100', DRIVER: 'latest' }
-              - { CUDA_VER: '11.8.0', LINUX_VER: 'ubuntu22.04', ARCH: 'arm64', PY_VER: '3.10', GPU: 'a100', DRIVER: 'latest' }
+              - { CUDA_VER: '12.0.1', LINUX_VER: 'ubuntu22.04', ARCH: 'amd64', PY_VER: '3.10', GPU: 'v100', DRIVER: 'latest' }
```
Comment (jakirkham) [inline .github/workflows/conda-cpp-tests.yaml:68]: It looks like the CUDA 11.8.0 ARM jobs are being dropped. Is that intended?

Also see this on subsequent lines below

Comment (bdice) [reply .github/workflows/conda-cpp-tests.yaml:69]: Good catch. Originally @ajschmidt8 had proposed to replace one ARM job for CUDA 11 with one ARM job for CUDA 12 to keep the CI matrix from ballooning. That was done in commits 229b0ebccb20d42bd35520c1d51e1f0cfc4f2c2e and 849a00e8e32aeb2a06a2e40e2def8b0a583529be. However, we had to remove CUDA 12 ARM support due to conda-forge glibc support reasons. I removed the "replacement job" in da85c6e602f0f532b07c7d89947a157406a1c17e but didn't add back the original job. I have fixed that in 78e880159d158707e4632b96d29fbb4e7e75d01a.

File: .github/workflows/conda-cpp-tests.yaml [hunk 2]
```diff
@@ -78,7 +79,8 @@ jobs:
               - { CUDA_VER: '11.8.0', LINUX_VER: 'ubuntu22.04', ARCH: 'amd64', PY_VER: '3.10', GPU: 'v100', DRIVER: 'latest' }
               - { CUDA_VER: '11.8.0', LINUX_VER: 'ubuntu22.04', ARCH: 'arm64', PY_VER: '3.10', GPU: 'a100', DRIVER: 'latest' }
               - { CUDA_VER: '11.8.0', LINUX_VER: 'rockylinux8', ARCH: 'amd64', PY_VER: '3.10', GPU: 'v100', DRIVER: 'latest' }
-              - { CUDA_VER: '11.8.0', LINUX_VER: 'rockylinux8', ARCH: 'arm64', PY_VER: '3.10', GPU: 'a100', DRIVER: 'latest' }
+              - { CUDA_VER: '11.8.0', LINUX_VER: 'ubuntu22.04', ARCH: 'arm64', PY_VER: '3.10', GPU: 'a100', DRIVER: 'latest' }
```
Comment (bdice) [inline .github/workflows/conda-cpp-tests.yaml:82]: Match previous job settings.
```suggestion
              - { CUDA_VER: '11.8.0', LINUX_VER: 'rockylinux8', ARCH: 'arm64', PY_VER: '3.10', GPU: 'a100', DRIVER: 'latest' }
```

File: .github/workflows/conda-cpp-tests.yaml (hunk 1 above)
Comment (bdice) [reply .github/workflows/conda-cpp-tests.yaml:69]: ...and fixed again in 59da32dc4d6105f1a67a5bd728f6f9486359b18a. Resolving.

**Review by ajschmidt8 — APPROVED** [review]

File: .github/workflows/conda-cpp-tests.yaml (hunk 1 above)
Comment (jakirkham) [reply .github/workflows/conda-cpp-tests.yaml:69]: Thanks Bradley! 🙏

**Pull request merged**